pkg/source/builtin.go

```diff
@@ -0,0 +1,22 @@
+package source
+
+import (
+	"context"
+
+	"github.com/re-cinq/aether/pkg/config"
+	"github.com/re-cinq/aether/pkg/providers/gcp"
+	v1 "github.com/re-cinq/aether/pkg/types/v1"
+)
+
+func BuiltInSources(ctx context.Context) []Source {
+	var sources []Source
+
+	// TODO: not sure how to do this without if statements but feel this could
+	// look cleaner
+	if cfg, exists := config.AppConfig().Providers[v1.GCP]; exists {
+		for _, s := range gcp.Sources(ctx, &cfg) {
+			sources = append(sources, s)
+		}
+	}
```

Comment:
Not uber happy with this :sweat_smile: looks dirty but not sure how else to make it work